.github/workflows/build-tests.yml

```
@@ -0,0 +1,51 @@
+name: Build-Testing
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+
+    steps:
+    - name: Install Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y cmake gcovr
+
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Configure CMake
+      working-directory: ./IOT
+      run: cmake -B ${{github.workspace}}/IOT/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DBUILD_TEST_SUITE=TRUE -DENABLE_TEST_COVERAGE=TRUE
+
+    - name: Build
+      working-directory: ./IOT
+      run: cmake --build ${{github.workspace}}/IOT/build --config ${{env.BUILD_TYPE}} -j
```

This is not needed for this tiny project, but using all ressources provided by the runner may speed up workflow time ([Github linux runners have 4 cores!](https://docs.github.com/fr/actions/using-jobs/choosing-the-runner-for-a-job#standard-github-hosted-runners-for-public-repositories)).
But we may continue copy paste this file for other projects 👀.

```suggestion
      run: cmake --build ${{github.workspace}}/IOT/build --config ${{env.BUILD_TYPE}} -j$(nproc)
```